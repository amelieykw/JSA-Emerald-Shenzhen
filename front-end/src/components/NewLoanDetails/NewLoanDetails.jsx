import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { Text, View, Content, Icon, Container } from 'native-base'
import style from './NewLoanDetailsStyle'
import LoanPickers from './LoanPickers'
import data from '../../../helpers/mockData_FE'
import FloatingButton from '../FloatingButton/FloatingButton'

export default function NewLoanDetails({ navigation }) {
  const [receivingAccount, setreceivingAccount] = useState(data.receivingAccount[0])
  const [defaultPayment, setdefaultPayment] = useState(data.defaultPayment[0])

  const amount = navigation.getParam('amount')

  const handlePress = () => {
    // eslint-disable-next-line no-undef
    alert('oh shit!! you pressed me!!')
  }
  return (
    <Container>
      <Content style={style.content}>
        <View style={style.upDetailsView}>
          <Text style={style.amountFont}>{`$ ${amount}`}</Text>
          <Text style={style.transactionFont}>will be added to your</Text>
          <Text style={style.selectionTitle}>Receiving amount</Text>
          <LoanPickers
            selected={receivingAccount}
            setSelected={setreceivingAccount}
            itemList={data.receivingAccount}
          />
          <Text style={style.transactionFont}>and you will pay</Text>
          <Text style={style.amountFont}>$ 86.84</Text>
          <Text style={style.selectionTitle}>Default Payment</Text>
          <LoanPickers
            selected={defaultPayment}
            setSelected={setdefaultPayment}
            itemList={data.defaultPayment}
          />
        </View>
        <View style={style.downTipView}>
          <Icon name="exclamation-circle" type="FontAwesome" />
          <Text style={style.exclamationText}>
            {`This amount will be automatically removed from your account every ${defaultPayment.value}.`}
          </Text>
        </View>
      </Content>
      <FloatingButton
        buttonStyle={{ width: 130, backgroundColor: '#72bb53' }}
        icon={{ type: 'MaterialIcons', name: 'done' }}
        text="Finish"
        handlePress={handlePress}
      />
    </Container>
  )
}
NewLoanDetails.navigationOptions = {
  title: 'New Loan Details'
}

NewLoanDetails.propTypes = {
  navigation: PropTypes.objectOf(PropTypes.any)
}

NewLoanDetails.defaultProps = {
  navigation: undefined
}
