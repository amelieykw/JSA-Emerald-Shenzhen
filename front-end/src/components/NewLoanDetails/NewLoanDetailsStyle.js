import { StyleSheet } from 'react-native'
import { Platform } from '@unimodules/core'

const style = StyleSheet.create({
  content: {
    padding: 10,
    height: '100%'
  },
  upDetailsView: {
    padding: 10
  },
  picker: {
    borderBottomColor: '#E8EAF6',
    borderBottomWidth: 1
  },
  pickerItem: {
    fontSize: 30
  },
  downTipView: {
    backgroundColor: '#E8EAF6',
    paddingHorizontal: 35,
    paddingVertical: Platform.OS === 'android' ? 20 : 30,
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  amountFont: {
    fontSize: 48
  },
  selectionTitle: {
    fontSize: 14,
    marginTop: 20
  },
  exclamationText: { fontWeight: '200', color: '#7C7D83', marginLeft: 20 },
  transactionFont: { fontSize: 20, marginTop: 25 }
})
export default style
