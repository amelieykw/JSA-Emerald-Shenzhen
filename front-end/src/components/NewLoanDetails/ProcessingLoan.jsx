import React from 'react'
import PropTypes from 'prop-types'
import { Text, View, Content, Container } from 'native-base'
// import FloatingButton from '../FloatingButton/FloatingButton'

export default function ProcessingLoan({ navigation }) {
  // const handlePress = () => {
  //   navigation.navigate('LoanList', {
  //     productId: navigation.getParam('productId'),
  //     amount: navigation.getParam('amount'),
  //     duration: navigation.getParam('duration'),
  //     monthlyPay: navigation.getParam('monthlyPay'),
  //     receivingAccount: navigation.getParam('receivingAccount'),
  //     defaultPayment: navigation.getParam('defaultPayment')
  //   })
  // }
  return (
    <Container>
      <Content>
        <View>
          <Text>{`$ ${navigation.getParam('amount')}.`}</Text>
        </View>
      </Content>
    </Container>
  )
}
ProcessingLoan.navigationOptions = {
  title: 'New Loan Details'
}

ProcessingLoan.propTypes = {
  navigation: PropTypes.objectOf(PropTypes.any)
}

ProcessingLoan.defaultProps = {
  navigation: undefined
}
