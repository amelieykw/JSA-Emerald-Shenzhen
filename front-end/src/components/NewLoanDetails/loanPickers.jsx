import React from 'react'
import { Form, Picker, Icon } from 'native-base'
import PropTypes from 'prop-types'
import style from './NewLoanDetailsStyle'

export default function LoanPickers({ selected, setSelected, itemList }) {
  return (
    <>
      <Form style={style.picker}>
        <Picker
          textStyle={style.pickerItem}
          itemStyle={style.pickerItem}
          mode="dropdown"
          selectedValue={selected}
          onValueChange={value => setSelected(value)}
          iosIcon={<Icon name="arrow-down" />}
        >
          {itemList.map(item => {
            return <Picker.Item key={item.id} label={item.label} value={item} />
          })}
        </Picker>
      </Form>
    </>
  )
}
LoanPickers.propTypes = {
  selected: PropTypes.objectOf(PropTypes.any).isRequired,
  setSelected: PropTypes.func.isRequired,
  itemList: PropTypes.arrayOf(PropTypes.any).isRequired
}
