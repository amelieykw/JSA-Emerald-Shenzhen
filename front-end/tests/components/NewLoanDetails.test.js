import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import NewLoanDetails from '../../src/components/NewLoanDetails/NewLoanDetails'

Enzyme.configure({ adapter: new Adapter() })

describe('test NewLoan View', () => {
  it('testing NewLoanDetails have 1 children, 2 grant children and match snapshot', () => {
    const warpper = shallow(<NewLoanDetails />)
    expect(warpper.children.length).toBe(1)
    expect(warpper.children().children().length).toBe(2)
    expect(warpper).toMatchSnapshot()
  })
})
