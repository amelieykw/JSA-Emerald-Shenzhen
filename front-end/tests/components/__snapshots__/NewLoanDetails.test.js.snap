// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test NewLoan View testing NewLoanDetails have 1 children, 2 grant children and match snapshot 1`] = `ShallowWrapper {}`;
